lib/Intlc/Parser.hs

```diff
@@ -149,15 +133,13 @@ interp = do
           , Number <$ string "number"
           , Date <$> (string "date" *> sep *> dateTimeFmt)
           , Time <$> (string "time" *> sep *> dateTimeFmt)
-          , try $ uncurry Select <$> (string "select" *> sep *> selectCases)
           , Plural <$> withPluralCtx n (
                   string "plural" *> sep *> cardinalPluralCases
               <|> string "selectordinal" *> sep *> ordinalPluralCases
             )
+          , uncurry Select <$> (string "select" *> sep *> selectCases)
           ]
-        withPluralCtx n p = pushPluralCtx n *> p <* popPluralCtx
-        pushPluralCtx = modify . pushPluralArgName
-        popPluralCtx = modify popPluralArgName_
+        withPluralCtx n p = withReaderT (const . ParserState . pure $ n) p
```

Comment:
The `const` in `withReaderT (const f)` makes me think I'm missing something. :thinking: